crates/secrets-client/src/lib.rs

```diff
@@ -0,0 +1,176 @@
+use async_nats::HeaderMap;
+use nkeys::XKey;
+use wasmcloud_secrets_types::{
+    Secret, SecretRequest, SecretResponse, RESPONSE_XKEY, WASMCLOUD_HOST_XKEY,
+};
+
+const DEFAULT_API_VERSION: &str = "v0";
+
+#[derive(Debug)]
+pub enum SecretClientError {
```

Comment:
Super Nit: I'd just recommend using `thiserror` rather than manually implementing display

Reply:
I was hoping to hold the nits for the `secrets-client` crate for a followup since there are a couple of things we'd like to improve, like Victor's comments that are captured in #2411 . Do you have an aversion to that?

Reply:
Nope that is fine